lib/improver/cli/apply_emos_coefficients.py

```diff
@@ -33,150 +33,22 @@
-        action='store_true',
-        help='If True, where the percentiles exceed the ECC bounds range, '
-             'raise a warning rather than an exception. This occurs when the '
-             'current forecast is in the form of probabilities and is '
-             'converted to percentiles, as part of converting the input '
-             'probabilities into realizations.')
-    parser.add_argument(
-        '--predictor_of_mean', metavar='PREDICTOR_OF_MEAN',
-        choices=['mean', 'realizations'], default='mean',
-        help='String to specify the predictor used to calibrate the forecast '
-             'mean. Currently the ensemble mean ("mean") and the ensemble '
-             'realizations ("realizations") are supported as options. '
-             'Default: "mean".')
-    parser.add_argument(
-        '--landsea_mask', metavar="LANDSEA_MASK", default=None,
-        help="The netCDF file containing a land-sea mask on the same domain "
-             "as the forecast that is to be calibrated. Land points are "
-             "specified by ones and sea points are specified by zeros. "
-             "Supplying this file will enable land-only calibration, in which "
-             "sea points are returned without the application of calibration.")
-    parser.add_argument(
-        '--shape_parameters', metavar="SHAPE_PARAMETERS", nargs="*",
-        default=None, type=float,
-        help="The shape parameters required for defining the distribution "
-             "specified by the distribution argument. The shape parameters "
-             "should either be a number or 'inf' or '-inf' to represent "
-             "infinity. Further details about appropriate shape parameters "
-             "are available in scipy.stats. For the truncated normal "
-             "distribution with a lower bound of zero, as available when "
-             "estimating EMOS coefficients, the appropriate shape parameters "
-             "are 0 and inf.")
-
-    args = parser.parse_args(args=argv)
-
-    # Load Cubes
-    current_forecast = load_cube(args.forecast_filepath)
-    coeffs = load_cube(args.coefficients_filepath, allow_none=True)
-    landsea_mask = load_cube(args.landsea_mask, allow_none=True)
-    # Process Cube
-    result = process(current_forecast, coeffs, landsea_mask,
-                     args.distribution, args.num_realizations,
-                     args.random_ordering, args.random_seed,
-                     args.ecc_bounds_warning, args.predictor_of_mean,
-                     args.shape_parameters)
-    # Save Cube
-    save_netcdf(result, args.output_filepath)
-
-
-def process(current_forecast, coeffs, landsea_mask, distribution,
-            num_realizations=None, random_ordering=False, random_seed=None,
-            ecc_bounds_warning=False, predictor_of_mean='mean',
-            shape_parameters=None):
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(current_forecast: cli.inputcube,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Too many branches (13/12)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4668197)

Reply:
OK, I'm wondering if there is potential to sort out some of the `if / else` logic in lines 173-207 (at a glance I can see at least one branch that could be simplified, and there are probably more).  This could reduce the number of both branches and local variables.